File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/merchants/MerchantsContract.java
package org.mifos.mobilewallet.mifospay.merchants;

import org.mifos.mobilewallet.core.data.fineract.entity.accounts.savings.SavingsWithAssociations;
import org.mifos.mobilewallet.mifospay.base.BasePresenter;
import org.mifos.mobilewallet.mifospay.base.BaseView;

import java.util.List;

public interface MerchantsContract {
    interface MerchantsPresenter extends BasePresenter {

        void fetchMerchants();
    }

    interface MerchantsView extends BaseView<MerchantsPresenter> {

        void listMerchantsData(List<SavingsWithAssociations> savingsWithAssociationsList);

        void showErrorStateView(int drawable, int title, int subtitle);

        void showEmptyStateView();

        void showMerchants();

        void showMerchantFetchProcess();

    }
}

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/merchants/MerchantsContract.kt
package org.mifos.mobilewallet.mifospay.merchants

import org.mifos.mobilewallet.core.data.fineract.entity.accounts.savings.SavingsWithAssociations
import org.mifos.mobilewallet.mifospay.base.BasePresenter
import org.mifos.mobilewallet.mifospay.base.BaseView

interface MerchantsContract {
    interface MerchantsPresenter : BasePresenter {
        fun fetchMerchants()
    }

    interface MerchantsView : BaseView<MerchantsPresenter?> {
        fun listMerchantsData(savingsWithAssociationsList: List<SavingsWithAssociations>?)
        fun showErrorStateView(drawable: Int, title: Int, subtitle: Int)
        fun showEmptyStateView()
        fun showMerchants()
        fun showMerchantFetchProcess()
    }
}